samples/UpdateSamples.ps1

```diff
@@ -0,0 +1,32 @@
+$gitHubUrl = 'https://github.com'
+$files = Get-ChildItem -Directory $PSScriptRoot | Get-ChildItem -File -Filter 'readme.md'
+foreach($file in $files)
+{
+    $fileContent = Get-Content $file
+    $branch = ($fileContent | Where-Object { $_ -like '*branch:*' } | Select-Object -First 1) -Split ':' | Select-Object -Last 1
+    $updatedLines = foreach($line in $fileContent)
+    {
+        if($line -like "*$gitHubUrl*")
+        {
+            $lineSplit = $line -Split '/'
+            $blobIndex = $lineSplit.IndexOf('blob')
+            $oldHash = $lineSplit[($blobIndex + 1)]
+
+            $org = $lineSplit[($blobIndex - 2)]
+            $repo = $lineSplit[($blobIndex - 1)]
+            $gitLink = "$gitHubUrl/$org/$repo.git"
+            $branchList = (git ls-remote $gitLink)
+            if($branch)
+            {
+                $branchList = $branchList | Where-Object { $_ -like "*$branch*" }
```

Comment:
This isn't full-proof since a branch name could exist within multiple branches. It'll end up taking the first branch that matches the search. But, I don't feel the need to make this more complex unless there is a specific issue with it.